E1_coldflow/brain_utils.h

```diff
@@ -16,15 +16,37 @@
 #include <string>
 #include <SdFat.h>
 #include <TimeLib.h>
+#include <vector>
 
 String make_packet(struct sensorInfo sensor);
 uint16_t Fletcher16(uint8_t *data, int count);
 void chooseValveById(int id, struct valveInfo *valve);
 
 const int numCryoTherms = 2;
-int cryoThermAddrs[numCryoTherms] = {0x60, 0x67}; //the second one is 6A or 6B, not sure which for Addr pin set to 1/2
+int cryoThermAddrs[numCryoTherms] = {0x60, 0x67};
 _themotype cryoTypes[numCryoTherms] = {MCP9600_TYPE_J, MCP9600_TYPE_T};
 
+const int numADCSensors = 2;
+int ADSAddrs[numADCSensors] = {0b1001010, 0b1001000};
+int adcDataReadyPins[numADCSensors] = {29, 28};
+ADS1219 ** ads;
+
+// sparse matrix representation of mapping
+int adc1ThermMap[] = {0, 0, 0, 0};
+int adc2ThermMap[] = {0, 0, 1, 0};
+int * adcThermMap[] = {adc1ThermMap, adc2ThermMap};
+
+std::vector<int> *analogThermADCMap;
+
+void initConfig() {
+  analogThermADCMap = (std::vector<int> *)malloc(numADCSensors * sizeof(std::vector<int>));
+  for (int i = 0; i < numADCSensors; i++) {
+    // initialize vector w/ iterator to beginning and end of array
+    analogThermADCMap[i] = std::vector<int>(adcThermMap[i], adcThermMap[i] + sizeof(adcThermMap[i]) / sizeof(int) );
+    analogThermADCMap[i].shrink_to_fit();
+  }
+}
+
```

Comment:
This will be the general layout of the config files. Open to ideas to make it cleaner, but given that everything here is by definition hardcoded, not sure what we can do here.

Reply:
What exactly is the advantage of using the vectors here? At the end of the day they're just used to iterate through each ADC map once inside the library, right?

Reply:
I think it could be simpler to pass an array to each type of sensor, cause then you could avoid the complexity & memory overhead of vectors. Instead of passing an array with pointers to each library, you could instead just have one global adc array, and pass to each library an array containing indices for that global array, and the channel number for each sensor. Something like this:


```
for (int i = 0; i < numADCSensors; i++) {
    ads[i] = new ADS1219(adcDataReadyPins[i], ADSAddrs[i], &Wire);
  }

int numTherm = 2;
int thermAdcs[] = {0, 1};
int thermAdcCh[] = {0, 3};

int numPT = 4;
int ptAdcs[] = {0, 0, 1, 1};
int ptAdcChs[] = {1, 2, 0, 1};
```
-----

Then in the library itself you could just iterate over the # of sensors:

```
init (int numSensors, int *adcs, int * chs) {
      _numsensors = numSensors;
      _adcs = ADCs;
      _chs = chs;
      
}

read_data {
      for (int i = 0; i < _numSensors; i++) {
            rawRead = _adcs[i]->readData(chs[i]);
            /*Sensor Specific stuff*/
      }
}
```

Any thoughts/hesitations with an approach like this?

Reply:
nice! I was originally trying something like this, but I kept trying to keep the map 2d and was having issues. Didn't think about flattening it like this. Will implement and rerequest review